src/frontend/src/app/AppAuthenticated.tsx

```diff
@@ -45,7 +45,8 @@ const AppAuthenticated: React.FC<AppAuthenticatedProps> = ({ userId }) => {
   };
 
   if (isLoading || !userSettingsData) return <LoadingIndicator />;
-  if (isError) return <ErrorPage error={error} message={error.message} />;
+
+  if (isError && error) return <SessionTimeoutAlert />;
```

Comment:
could you check that the error message is the same one we're expecting from the backend, and if so, display the SessionTimeoutAlert and otherwise display the normal error page. You should be able to just make that endpoint in the backend throw that error to manually test it